Added realisation class Distance

SharonLewislxz48 opened the pull request.

SharonLewislxz48 committed:
Added realisation class Distance

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,3 +1,70 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km):
```
The checklist requires all methods to have type annotations. For example, this method signature should be `def __init__(self, km: int | float) -> None:`. Please add appropriate annotations to all methods in this class. You will also need to add `from __future__ import annotations` at the top of the file to use `Distance` as a type hint for other methods.

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,3 +1,70 @@
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
```
This `else` statement violates checklist item 'Make Code Easier #1': 'Avoid unnecessary `else` statements'. Consider how you could restructure the method to avoid it. Often, this can be achieved by returning from within the `if`/`elif` blocks.

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,3 +1,70 @@
+            return NotImplemented
+        return self
+
+    def __mul__(self, other):
```
This method is also missing type annotations. The checklist specifies that `__mul__` and `__truediv__` should only accept a number for the `other` argument. The correct signature would be `def __mul__(self, other: int | float) -> Distance:`.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Great job on the functional implementation of the `Distance` class! All the arithmetic and comparison logic works correctly. However, changes are needed to meet the project's core requirements as outlined in the checklist.

Specifically, two items need your attention:

1.  **Missing Type Annotations**: All methods lack type annotations for their arguments and return values. For example, the `__init__` method signature should be `def __init__(self, km: int | float) -> None:`. Please add these to all methods and include `from __future__ import annotations` at the top of the file. [CHECKLIST ITEM #1]
2.  **Unnecessary `else` Statement**: The `__iadd__` method includes an `else` statement that violates the checklist's guideline for simpler code. This can be resolved by moving the `return self` statement outside of the conditional block.

Please address these points to align your code with the project standards.

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

SharonLewislxz48 committed:
Add type hints to Distance class mAdd type annotations and fix __iadd__ methodethods

- Added type annotations to all methods
- Added 'from __future__ import annotations' at the top
- Removed unnecessary else block from __iadd__ method

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Excellent work on this submission! You've successfully addressed all the feedback from the previous review. Your implementation now meets all the requirements, including the correct use of type annotations and the removal of the unnecessary `else` statement. The code is clean, correct, and functions perfectly. I am happy to approve your solution. Well done!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
